numba/cuda/decorators.py

```diff
@@ -71,8 +70,37 @@ def jit(func_or_sig=None, device=False, inline=False, link=[], debug=None,
         msg = _msg_deprecated_signature_arg.format('bind')
         raise DeprecationError(msg)
 
+    debug = config.CUDA_DEBUGINFO_DEFAULT if debug is None else debug
     fastmath = kws.get('fastmath', False)
-    if not sigutils.is_signature(func_or_sig):
+
+    if sigutils.is_signature(func_or_sig):
```

Comment:
note: think this is just code motion, switching the if/else branches

Reply:
It also simplified the logic a little bit to turn it upside down - a couple of now-redundant checks are removed, but the remaining code is just moved jue to the inverted condition.